packages/pds/src/api/com/atproto/server/getSession.ts

```diff
@@ -14,12 +14,25 @@ export default function (server: Server, ctx: AppContext) {
     }),
     handler: async ({ auth, req }) => {
       if (ctx.entrywayAgent) {
-        return resultPassthru(
-          await ctx.entrywayAgent.com.atproto.server.getSession(
-            undefined,
-            ctx.entrywayPassthruHeaders(req),
-          ),
+        // Allow proxying of dpop bound requests by using service auth instead
+        const headers =
+          req.headers.authorization?.split(' ')[0].toLowerCase() === 'dpop'
```

Comment:
This check here is a bit dense.  Could we break it up a little, or perhaps use `credentials.type` if that's able to do the trick.

Reply:
💯